mimic/test/test_identity_osksadm.py

```diff
@@ -0,0 +1,366 @@
+from mimic.test.dummy import make_example_external_api, ExternalApiStore
+from mimic.test.helpers import json_request, request
+
+
+@ddt.ddt
+class TestIdentityMimicOSKSCatalogAdminListExternalServices(SynchronousTestCase):
+    """
+    Tests for ``/identity/v2.0/services``, provided by
+    :obj:`mimic.rest.idenity_api.IdentityApi`
+    """
+    def setUp(self):
+        self.core = MimicCore(Clock(), [])
+        self.root = MimicRoot(self.core).app.resource()
+        self.uri = "/identity/v2.0/services"
+        self.eeapi_name = u"externalServiceName"
+        self.headers = {
+            b'X-Auth-Token': [b'ABCDEF987654321']
+        }
+        self.verb = b"GET"
+
+    def test_auth_fail(self):
+        """
+        GET with no X-Auth-Token header results in 401.
+        """
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri))
+
+        self.assertEqual(response.code, 401)
+        self.assertEqual(json_body['unauthorized']['code'], 401)
+
+    @ddt.data(
+        0, 1, 10
+    )
+    def test_listing(self, api_entry_count):
+        """
+        GET will list the registered services.
+        """
+        api_list = []
+
+        # create the desired number of services per test parameter
+        for _ in range(api_entry_count):
+            api_list.append(
+                ExternalApiStore(
+                    text_type(uuid.uuid4()),
+                    self.eeapi_name + text_type(uuid.uuid4()),
+                    'service-' + text_type(uuid.uuid4()),
+                )
+            )
+
+        # add the services
+        for api in api_list:
+            self.core.add_api(api)
+
+        # retrieve the listing using the REST interface
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri,
+                         headers=self.headers))
+
+        def validate_api(api_id, api_type, api_name):
+            """
+            Lookup the API in the test's set of APIs  and match the values
+            """
+            for api in api_list:
+                if api.uuid_key == api_id:
+                    # service found, check values and return
+                    self.assertEqual(api_id, api.uuid_key)
+                    self.assertEqual(api_type, api.type_key)
+                    self.assertEqual(api_name, api.name_key)
+                    return
+
+            # no service found, raise assertion error
+            self.assertFalse(
+                True,
+                "Unknown service: {0} - {1} - {2}".format(
+                    api_id, api_type, api_name))
+
+        self.assertEqual(response.code, 200)
+        self.assertEqual(len(json_body["OS-KSADM:services"]), len(api_list))
+        # ensure all services in the response match one in the generated
+        # initially generated set
+        for entry in json_body["OS-KSADM:services"]:
```

Comment:
Yes, it means no APIs have been registered. I believe that is part of the test via `ddt`; though can't remember if it's in this specific test or not.